### ISSUE TITLE
jobs dashboard only displays "Not a Registered Service"

### ISSUE THREAD

**Issue by lvmajor** [jamesmh/coravelpro#16]
As the title says, after performing the following steps:
1. Using Asp.Net Core and EF Core 2.2
2. Installed both Coravel (3.0.0) and Coravel.Pro(1.2.1) NuGet packages
3. Added `services.AddCoravelPro(typeof(ApplicationDbContext));` to the Startup `ConfigureServices` method (where the db context name is effectively `ApplicationDbContext`)
4. Added `app.UseCoravelPro();` to Startup `Configure` method (tried both after and before `UseMvc`
6. Made my DbContext implement the `ICoravelProDbContext` and created/applied related migration (did verify the tables are present in the database)
7. Implemented an `Coravel.Invocable.IInvocable` and added it to the services container as a `Transient` dependency
8. Started the app, hoping for the best.... 

Yet, when accessing the dashboard's Jobs page, under the Jobs section, it only displays "Not a Registered Service" on the right column under the "Execute" header

![image](https://user-images.githubusercontent.com/1885400/65690189-7e1cff80-e03c-11e9-9463-02d95741534e.png)

I'd like to know if you would have an idea of what can cause this message to appear :)

Thanks in advance and have a great day!

**Comment by jamesmh** [comment]
For some reason the Invocable isn't appearing in that list... are your invocables in another assembly or in the same one as your web application?

**Comment by lvmajor** [comment]
They are in the same assembly... that's weird. Is the code accessible for this part so that I can try to figure out what happens? Or is the pro features code private? (which I would understand :) )

**Comment by jamesmh** [comment]
No, the code not accessible. I know other people have been having issues with .NET Core 3 and ordering of the `UseX` methods. What version of .NET core are you using?

**Comment by lvmajor** [comment]
still on 2.2 for now as there are some changes that break my app with new endpoint routing, did not update yet to 3.0 because of that.

I guess the problem here has to do with some other services or ordering in the Configure method of the Startup class. I checked the logs but I can't find anything that points me in the right direction...  I tried putting Coravel before or after Mvc to no avail... 

I will see if I can manage to make it work during the weekend

**Comment by jamesmh** [comment]
If you can send the code you have for the configure and configureServices methods I might be able to pin point where the issue might be?

**Comment by lvmajor** [comment]
Sorry for the delay, I'll see if I can provide a repro tonight. I will also try to remove some of the configured services in the `Configure` method and add them back one by one until it fails again to help pinpoint the error (if it's caused by an incompatibility between middlewares/services)

**Comment by jamesmh** [comment]
👍

**Comment by lvmajor** [comment]
Again, sorry for taking so long, I've been pretty busy during the week with my job. After investigating, I found out the problematic code seems to be the following:
```
.AddJsonOptions(options =>  // Chained after services.AddMvc()....
{
    options.SerializerSettings.ContractResolver = new DefaultContractResolver();    // This switches back from camelCase to PascalCase for json serialization (to comply with default .NET models casing)
    options.SerializerSettings.DateFormatString = "yyyy-MM-dd";
});
```
In fact, as soon as I comment the first line in this config snippet (the default contract resolver), the problem goes away and Coravel Dashboard and jobs pages start working again.

**Comment by set-webmaster** [comment]
@jamesmh seeing this, I guess your dashboard depends on camel casing of variables returned from Json and that's why it fails?

Would you have an idea of how to make it work with both camel and Pascal casing?

**Comment by jamesmh** [comment]
Ya, for that to work Coravel Pro's code would need to change and use it's own custom json formatter? But as it stands, sounds like it'll only work using the default casing.

**Comment by lvmajor** [comment]
Okay, well I guess it could be documented somewhere just to inform potential users of that behavior. As  IIRC, there was a lot of people using PascalCasing serialization (which was the default behavior in previous versions of asp.net core), so for those like me who opted to keep the `DefaultContractResolver` to prevent breaking pieces of code depending on pascal casing, Coravel Pro is not compatible.
